eacl/table.go

```diff
@@ -1,28 +1,71 @@
 package eacl
 
 import (
-	"crypto/sha256"
+	"errors"
 	"fmt"
 
-	v2acl "github.com/nspcc-dev/neofs-api-go/v2/acl"
+	"github.com/nspcc-dev/neofs-api-go/v2/acl"
 	"github.com/nspcc-dev/neofs-api-go/v2/refs"
 	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
 	"github.com/nspcc-dev/neofs-sdk-go/version"
 )
 
-// Table is a group of ContainerEACL records for single container.
+// Table represents NeoFS extended ACL (eACL): group of rules managing access to
+// NeoFS resources in addition to the basic ACL.
 //
-// Table is compatible with v2 acl.EACLTable message.
+// See also [acl.Basic].
 type Table struct {
-	version version.Version
+	version *version.Version
 	cid     *cid.ID
 	records []Record
 }
 
+// New constructs eACL from the given list of access rules. Being applied as
+// part of access control to the NeoFS resources, the rules are matched
+// according to the first hit: if a rule with a certain index is applicable,
```

Comment:
do we want this behavior? i would say that looking for every applicable rule and choosing the most restrictive looks better